src/UrlParams.ts

```diff
@@ -258,6 +260,8 @@ export const getUrlParams = (
   search = window.location.search,
   hash = window.location.hash,
 ): UrlParams => {
+  // Replace any non-printable characters that another client may have inserted.
+  search = search.replaceAll(/^[^ -~]+|[^ -~]+$/g, "");
```

Comment:
Looks like you're doing this replacement twice as ParamParser is invoked just after this.

Also, we do want to allow non-ASCII characters in the URL parameters in general. For example `displayName` or `homeserver` could contain all kinds of Unicode characters and it's valid for them to appear at the end of the search string.